feat: add support for direnv to second line

patrickserrano opened the pull request:
Add support for displaying the current directory's Python virtual environment when [direnv](https://direnv.net) and [pyenv](https://github.com/pyenv/pyenv) are installed and being used.

<img width="841" alt="image" src="https://user-images.githubusercontent.com/1302453/83426876-d42d1880-a3fd-11ea-8fd7-499805d0d114.png">

sjk07 commented:
@patrickserrano cool! does this also work in sh?

also I don't use python but is this all the information that would be required for a user to understand what virtual env they are looking in and is there a better way to format the text? 

ex `(env-name python-version)`

finally, could you add some description to the readme

patrickserrano commented:
@sjk07 the name of the virtualenv shown in the prompt is name the user set when during virtualenv configuration + `-<python version>`. I think that's enough information to understand what they're looking at.

So in my screenshot the name of the virtualenv is `virtual-env-test` and it's using Python `3.6.5` which is how you get `virtual-env-test-3.6.5`. But when working on airflow the name that's displayed is `airflow-3.6.5`. 

As for `sh` support, do the interactive features of prompt work with `sh`? I don't use it, but when I switched to `sh` and sourced the profile, it said I was using prompt but I only had a single line prompt with none of the features I'd get in bash or zsh. I also don't see in where in `src/sh/profile` we would be setting the prompt display.

Also, updated readme.

sjk07 commented on README.md:
```diff
@@ -121,6 +122,93 @@ In order to test the installation routine for promptMastermind, we use a set of
+
+### Colors
+
+Glocal colors defined by promotMastermind:
```
Global =]

patrickserrano replied:
```suggestion
Global colors defined by promotMastermind:
```

dmccaffery commented on README.md:
````diff
@@ -121,6 +122,93 @@ In order to test the installation routine for promptMastermind, we use a set of
+Install [minikube][minikube-url]:
+
+```sh
+minikube-install
````
I would favour kind over minikube these days -- prompt does both; I left minikube in for backward compatibility reasons.

patrickserrano replied:
updated :)

dmccaffery commented on src/zsh/scripts/eval/set-prompt:
```diff
@@ -27,14 +27,26 @@ __am_prompt_emit_prompt_arrow() {
+__am_prompt_show_virtualenv() {
+    if [[ -n "$VIRTUAL_ENV" && -n "$DIRENV_DIR" ]]; then
+        setopt PROMPT_SUBST
+        export DIR_ENV=" ($(basename $VIRTUAL_ENV))"
```
avoid the export if possible. AM_PROMPT_ARROW is exported because some folks asked to be able to change the arrow to something they liked, which is supported via edit-vars.

Also, I would try to use a bash built-in to derive the base name, something like:

DIR_ENV=${VIRTUAL_ENV%%/*}

basename forks a process and will be slower to execute.

patrickserrano replied:
Updated to use a built-in instead of basename.

I moved the function to an eval for bash. Based on the documentation for direnv, if it's not a function it won't be seen be by the sub-bash process.

dmccaffery commented on src/bash/scripts/eval/set-prompt:
```diff
@@ -26,6 +26,12 @@ __am_prompt_emit_prompt_arrow() {
 
+__am_prompt_show_virtual_env() {
+  if [[ -n "$VIRTUAL_ENV" && -n "$DIRENV_DIR" ]]; then
+    echo "($(basename $VIRTUAL_ENV)) "
```
spacing is different between this and the zsh version (this has a space after while zsh version has a space before). Not sure if that's intended.

dmccaffery commented on README.md:
````diff
@@ -121,6 +122,93 @@ In order to test the installation routine for promptMastermind, we use a set of
+
+Flush DNS:
+
+```she
````
typo: should be "sh" -- also, flush-dns is macOS only -- might want to say that. :)

dmccaffery replied:
nvm; you did! lol
